client/jetpack-connect/test/authorize.js

```diff
@@ -155,55 +179,81 @@ describe( 'JetpackAuthorize', () => {
 	} );
 
 	describe( 'shouldAutoAuthorize', () => {
-		test( 'should return true for sso', () => {
-			const renderableComponent = <JetpackAuthorize { ...DEFAULT_PROPS } />;
-			const component = shallow( renderableComponent );
-			component.instance().isSso = () => true;
-			const result = component.instance().shouldAutoAuthorize();
+		test( 'should authorize if isSso', () => {
+			const authorizeMock = jest.fn();
+
+			const authQuery = {
+				...DEFAULT_PROPS.authQuery,
+				from: 'sso',
+				clientId: APPROVE_SSO_CLIENT_ID,
+			};
+
+			renderWithRedux(
+				<JetpackAuthorize
+					{ ...DEFAULT_PROPS }
+					authorize={ authorizeMock }
+					authQuery={ authQuery }
+				/>
+			);
 
-			expect( result ).toBe( true );
+			expect( authorizeMock ).toHaveBeenCalled();
 		} );
 
-		test( 'should return true for woo services', () => {
-			const renderableComponent = <JetpackAuthorize { ...DEFAULT_PROPS } />;
-			const component = shallow( renderableComponent );
-			component.setProps( {
-				authQuery: {
-					...DEFAULT_PROPS.authQuery,
-					from: 'woocommerce-services-auto-authorize',
-				},
-			} );
-			const result = component.instance().shouldAutoAuthorize();
+		test( 'should auto-authorize for WOO services', () => {
+			const authorizeMock = jest.fn();
+
+			const authQuery = {
+				...DEFAULT_PROPS.authQuery,
+				from: 'woocommerce-services-auto-authorize',
+			};
 
-			expect( result ).toBe( true );
+			renderWithRedux(
+				<JetpackAuthorize
+					{ ...DEFAULT_PROPS }
+					authorize={ authorizeMock }
+					authQuery={ authQuery }
+				/>
+			);
+
+			expect( authorizeMock ).toHaveBeenCalled();
 		} );
 
-		test( 'should return false for woocommerce onboarding', () => {
-			const renderableComponent = <JetpackAuthorize { ...DEFAULT_PROPS } />;
-			const component = shallow( renderableComponent );
-			component.setProps( {
-				authQuery: {
-					...DEFAULT_PROPS.authQuery,
-					from: 'woocommerce-onboarding',
-				},
-			} );
-			const result = component.instance().shouldAutoAuthorize();
+		test( 'should not auto-authorize for WOO onboarding', () => {
+			const authorizeMock = jest.fn();
+
+			const authQuery = {
+				...DEFAULT_PROPS.authQuery,
+				from: 'woocommerce-onboarding',
+			};
 
-			expect( result ).toBe( false );
+			renderWithRedux(
+				<JetpackAuthorize
+					{ ...DEFAULT_PROPS }
+					authorize={ authorizeMock }
+					authQuery={ authQuery }
+				/>
+			);
+
+			expect( authorizeMock ).not.toHaveBeenCalled();
 		} );
 
-		test( 'should return true for the old woocommerc setup wizard', () => {
-			const renderableComponent = <JetpackAuthorize { ...DEFAULT_PROPS } />;
-			const component = shallow( renderableComponent );
-			component.setProps( {
-				authQuery: {
-					...DEFAULT_PROPS.authQuery,
-					from: 'woocommerce-setup-wizard',
-				},
-			} );
-			const result = component.instance().shouldAutoAuthorize();
+		test( 'should auto-authorize for the old WOO setup wizard', () => {
+			const authorizeMock = jest.fn();
```

Comment:
Optionally we could create this mock on `beforeAll()` and reset it on `afterEach()`, or just create it on `beforeEach()` to avoid boilerplate in every test.

Reply:
👍🏻